.azure/scripts/coverage.yml

```diff
@@ -0,0 +1,48 @@
+# This task produces the coverage reports and publishes them to codecov.
+steps:
+- task: UsePythonVersion@0
+  inputs:
+    versionSpec: '3.8'
+
+- template: jdk.yml
+  parameters:
+    version: 11
+
+- script: mvn -q -f project/coverage package
+- script: |
+    python setup.py test_java
+    pip install gcovr pytest-cov jedi
+    pip install -r test-requirements.txt
+    pip install numpy typing_extensions
```

Comment:
I think it would be faster to pass all requirements in one pip invocation, but this is just a minor detail.

Reply:
Yeah I was debating this myself.  I was going to put it in doc-requirements.txt, but when I read the logs on lgtm I could see that it also is adding a bunch of stuff then pulling doc-requirements.  If I put this in doc-requirements I may foul up some of the stuff they preinstall.  So we end up needing rtd-requirements.txt (used only on azure) with the stuff they put in and doc-requirements for all the stuff we are adding.   Is that a workable solution?

Reply:
When there are conflicts with RTD, we should avoid installing them solely on RTD and just install them on cI. How did this fire up on lgtm? I dont really understand.

Reply:
Sorry I meant to say RTD not LGTM.   In RTD they first pull the repo then install a set of requirments.  Then it pulls doc-requirements .   I don't want to bust the requirements.   So I need to have another text file with the list of the stuff RTD puts in place.